lib/graphql_includable/relay/edge_with_node_connection_type.rb

```diff
@@ -0,0 +1,25 @@
+module GraphQLIncludable
+  module Relay
+    class EdgeWithNodeConnectionType
+      def self.create_type(
+        wrapped_type, edge_to_node_property:,
+        edge_type: wrapped_type.edge_type, edge_class: EdgeWithNode, nodes_field: GraphQL::Relay::ConnectionType.default_nodes_field, &block
+      )
+        custom_edge_class = edge_class
+
+        GraphQL::ObjectType.define do
+          name("#{wrapped_type.name}Connection")
+          description("The connection type for #{wrapped_type.name}.")
+          field :edges, types[edge_type], "A list of edges.", edge_class: custom_edge_class, property: :fetch_edges, edge_to_node_property: edge_to_node_property
```

Comment:
Redefining a Relay connection type is key to making this work. The trick is being able to call different `property` methods for `nodes` vs `edges` and tagging `edges` with how to get a `node`.